app/code/core/Mage/Core/Helper/EnvironmentConfigLoader.php

```diff
@@ -223,31 +248,54 @@ protected function setCache(Mage_Core_Model_Store $store, $value, string $path):
         $store->setConfigCache($configCache);
     }
 
+    /**
+     * @return array{array<int, string>, string}
+     * @throws Mage_Core_Exception
+     */
     protected function getConfigKey(string $configKey): array
     {
         $configKeyParts = array_filter(
             explode(
-                static::ENV_KEY_SEPARATOR,
+                self::ENV_KEY_SEPARATOR,
                 $configKey,
             ),
             'trim',
         );
-        [$unused, $scope] = $configKeyParts;
+
+        unset($configKeyParts[0]);
+
+        $scope = array_shift($configKeyParts);
+        if (!is_string($scope)) {
+            throw new Mage_Core_Exception(
+                'Invalid configuration key: ' . $configKey . '. Scope is not a string.',
+            );
+        }
+
         return [$configKeyParts, $scope];
     }
 
+    /**
+     * @throws Mage_Core_Exception
+     */
     protected function isConfigKeyValid(string $configKey): bool
     {
-        $sectionGroupFieldRegexp = sprintf('([%s]*)', implode('', static::ALLOWED_CHARS));
-        $allowedChars = sprintf('[%s]', implode('', static::ALLOWED_CHARS));
-        $regexp = '/' . static::ENV_STARTS_WITH . static::ENV_KEY_SEPARATOR . '(WEBSITES' . static::ENV_KEY_SEPARATOR
-            . $allowedChars . '+|DEFAULT|STORES' . static::ENV_KEY_SEPARATOR . $allowedChars . '+)'
-            . static::ENV_KEY_SEPARATOR . $sectionGroupFieldRegexp
-            . static::ENV_KEY_SEPARATOR . $sectionGroupFieldRegexp
-            . static::ENV_KEY_SEPARATOR . $sectionGroupFieldRegexp . '/';
+        $sectionGroupFieldRegexp = sprintf('([%s]*)', implode('', self::ALLOWED_CHARS));
+        $allowedChars = sprintf('[%s]', implode('', self::ALLOWED_CHARS));
+        $regexp = '/' . self::ENV_STARTS_WITH . self::ENV_KEY_SEPARATOR . '(WEBSITES' . self::ENV_KEY_SEPARATOR
+            . $allowedChars . '+|DEFAULT|STORES' . self::ENV_KEY_SEPARATOR . $allowedChars . '+)'
+            . self::ENV_KEY_SEPARATOR . $sectionGroupFieldRegexp
+            . self::ENV_KEY_SEPARATOR . $sectionGroupFieldRegexp
+            . self::ENV_KEY_SEPARATOR . $sectionGroupFieldRegexp . '/';
         // /OPENMAGE_CONFIG__(WEBSITES__[A-Z-_]+|DEFAULT|STORES__[A-Z-_]+)__([A-Z-_]*)__([A-Z-_]*)__([A-Z-_]*)/
 
-        return preg_match($regexp, $configKey);
+        $validatedConfigKey = preg_match($regexp, $configKey);
+        if ($validatedConfigKey === false) {
+            throw new Mage_Core_Exception(
```

Comment:
Why does it throw now? It not caught and should not throw imho.

Reply:
You are right, be we could log it?

Reply:
Hmm.. yea I could see that being useful.
That ENV key starts with the correct value, but somewhere down the line a typo happened. That could be useful to the developer.

Reply:
Remove it to avoid more complexity. Maybe change later?

Reply:
Remove what exactly?